sql/core/src/test/scala/org/apache/spark/sql/streaming/MultiStatefulOperatorsSuite.scala

```diff
@@ -878,6 +878,60 @@ class MultiStatefulOperatorsSuite
     testOutputWatermarkInJoin(join3, input1, -40L * 1000 - 1)
   }
 
+  // NOTE: This is the revise of the reproducer in SPARK-45637. CREDIT goes to @andrezjzera.
```

Comment:
IIRC, if we want to add credit information, we can amend the PR commits to include a co-author. This way, we don’t need to add the credit information in the code comments.

Reply:
Though the approach of "co-authorship" is not fine-grained one e.g. the credit is only for test, I can do that instead if it's more comfortable to people. Also I don't see we give CREDIT like this way, so probably the preference is what you proposed.